lib/puppet/file_serving/http_metadata.rb

```diff
@@ -21,6 +21,30 @@ def initialize(http_response, path = '/dev/null')
       @checksums[:md5] = "{md5}#{checksum}"
     end
 
+    if checksum = http_response['content-sha256']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksums[:sha256] = "{sha256}#{checksum}"
+    end
+
+    if checksum = http_response['content-sha384']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksums[:sha384] = "{sha384}#{checksum}"
+    end
+
+    if checksum = http_response['content-sha512']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksums[:sha512] = "{sha512}#{checksum}"
+    end
+
+    if checksum = http_response['content-sha224']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksums[:sha224] = "{sha224}#{checksum}"
+    end
+
```

Comment:
The Content-MD5 header is described in https://tools.ietf.org/html/rfc1864, but according to https://tools.ietf.org/html/rfc7231 it's now deprecated:

> The Content-MD5 header field has been removed because it was inconsistently implemented with respect to partial responses.

But I can't find any reference to other `Content-*` checksum types. Would we ever receive those responses?

Reply:
Right; however I was not sure whether it was intended that whoever writes a puppet file server on a http endpoint and if they were free to add any headers such as these (to convey the checksum).